src/Components/GEOSldas_GridComp/Shared/LDAS_TileCoordType.F90

```diff
@@ -214,7 +214,7 @@ function eq_grid_def_type( grid_1, grid_2 )
 
     type(grid_def_type), intent(in) :: grid_1, grid_2
     
-    if ( index(grid_1%gridtype,trim(grid_2%gridtype))>0        .and. &
+    if ( trim(grid_1%gridtype) == trim(grid_2%gridtype)        .and. &
```

Comment:
@weiyuan-jiang:  What's the motivation for this change?  I agree that this seems cleaner, but I'm wondering if the less stringent old formulation (i.e., `index(grid_1%gridtype,trim(grid_2%gridtype))>0` ) might have been chosen for a reason.  One example that should pass the old condition but not the new condition is the following:
```
grid_1%gridtype = SMAP_EASEv2
grid_2%gridtype = EASEv2
```
Could the change create problems with this kind of backward compatibility?

Reply:
What if grid1 and grid2 exchange the position?  I think the original one is incomplete.

Reply:
@weiyuan-jiang: Yes, the old formulation isn't symmetric.  Perhaps it's just an accidental miss of "trim" in the first argument of the index function, but what if this was done intentionally for some reason?  I suppose we can always make the change and do the usual tests.  If they work, we should be ok. 
If we accept this change, what's still needed in order to merge this PR?  Have you or @biljanaorescanin tested it after my latest changes (i.e., after commit https://github.com/GEOS-ESM/GEOSldas/pull/637/commits/bc854c2d7e4664164a0325a9c41a49e8e55fdc1d)?

Reply:
I'm testing it now should have update for tomorrow.